Source/Core/Core/FreelookManager.cpp

```diff
@@ -0,0 +1,222 @@
+
+    if (m_move_buttons->controls[2]->GetState<bool>())
+      g_freelook_camera.MoveHorizontal(g_freelook_camera.GetSpeed());
+
+    if (m_move_buttons->controls[3]->GetState<bool>())
+      g_freelook_camera.MoveHorizontal(-g_freelook_camera.GetSpeed());
+
+    if (m_move_buttons->controls[4]->GetState<bool>())
+      g_freelook_camera.MoveForward(g_freelook_camera.GetSpeed());
+
+    if (m_move_buttons->controls[5]->GetState<bool>())
+      g_freelook_camera.MoveForward(-g_freelook_camera.GetSpeed());
+
+    if (m_fov_buttons->controls[0]->GetState<bool>())
+      g_freelook_camera.IncreaseFovX(g_freelook_camera.GetFovStepSize());
+
+    if (m_fov_buttons->controls[1]->GetState<bool>())
+      g_freelook_camera.IncreaseFovX(-1.0f * g_freelook_camera.GetFovStepSize());
+
+    if (m_fov_buttons->controls[2]->GetState<bool>())
+      g_freelook_camera.IncreaseFovY(g_freelook_camera.GetFovStepSize());
+
+    if (m_fov_buttons->controls[3]->GetState<bool>())
+      g_freelook_camera.IncreaseFovY(-1.0f * g_freelook_camera.GetFovStepSize());
+
+    if (m_speed_buttons->controls[0]->GetState<bool>())
+      g_freelook_camera.ModifySpeed(1.0f / 1.1f);
+
+    if (m_speed_buttons->controls[1]->GetState<bool>())
+      g_freelook_camera.ModifySpeed(1.1f);
+
+    if (m_speed_buttons->controls[2]->GetState<bool>())
+      g_freelook_camera.ResetSpeed();
+
+    if (m_other_buttons->controls[0]->GetState<bool>())
+      g_freelook_camera.Reset();
+  }
+
+  if (m_other_buttons->controls[1]->GetState<bool>())
+  {
+    if (!m_toggle_pressed_last)
+    {
+      const Freelook::ControlType current_control_type =
+          Freelook::g_ActiveConfig.cameraConfig.iControlType;
+      if (current_control_type != Freelook::ControlType::None)
+        g_freelook_camera.ToggleEnabled();
+      OSD::AddMessage(
+          StringFromFormat("Freelook: %s", g_freelook_camera.IsActive() ? "Enabled" : "Disabled"));
+    }
+    m_toggle_pressed_last = true;
+  }
+  else
+  {
+    m_toggle_pressed_last = false;
+  }
+}
+
+namespace Freelook
+{
+static InputConfig s_config("FreelookController", _trans("Freelook"), "FreelookController");
+InputConfig* GetConfig()
+{
+  return &s_config;
+}
+
+void Shutdown()
+{
+  s_config.UnregisterHotplugCallback();
+
+  s_config.ClearControllers();
+}
+
+void Initialize()
+{
+  if (s_config.ControllersNeedToBeCreated())
+  {
+    s_config.CreateController<FreelookController>(0);
+  }
+
+  s_config.RegisterHotplugCallback();
+
+  Freelook::g_Config.Refresh();
+
+  s_config.LoadConfig(true);
+}
+
+void LoadConfig()
+{
+  s_config.LoadConfig(true);
+}
+
+bool IsInitialized()
+{
+  return !s_config.ControllersNeedToBeCreated();
+}
+
+ControllerEmu::ControlGroup* GetGroup(int pad_num, FreelookGroup group)
+{
+  return static_cast<FreelookController*>(s_config.GetController(pad_num))->GetGroup(group);
+}
+
+void Update()
+{
+  static_cast<FreelookController*>(s_config.GetController(0))->Update();
```

Comment:
Technically, shouldn't this update _all_ controllers?
Since we support multiple controllers (but only add _one_ in `Initialize`), I think we should either make this future-proof by just looping (if you can) or leaving a note that those two should be synchronized (if you can't).

Thoughts?

Reply:
Good point.  My plan was to update this when I added more controller support but there's no reason I couldn't loop right now.